### PR TITLE
Show session expired prompt

### PR DESCRIPTION
**Pull request by kaushiksaliya**
- Show session expired prompt.
- Manage get address timeout error.
- Remove space from text field when use add only space before text and click on send button and again tap on text field then space still there.

[Screencast from 13-08-24 04:32:15 PM IST.webm](https://github.com/user-attachments/assets/b7f1788b-c165-49c8-aa98-2746ce528d17)

[Fix textfield.webm](https://github.com/user-attachments/assets/f815bf60-a3e7-43ed-9565-8580fe7566b8)

### CODE REVIEW

File: app/lib/ui/flow/home/home_screen_viewmodel.dart [hunk 1]
```diff
@@ -169,6 +179,28 @@ class HomeViewNotifier extends StateNotifier<HomeViewState> {
       );
     }
   }
+
+  void listenUserSession(String userId, String sessionId) async {
+    try {
+      userService.getUserSessionByIdStream(userId, sessionId).listen((session) {
```
Comment (cp-ishita-g) [inline app/lib/ui/flow/home/home_screen_viewmodel.dart:185]: rename to `streamUserSessionById`

File: app/lib/ui/flow/home/home_screen_viewmodel.dart [hunk 2]
```diff
@@ -169,6 +179,28 @@ class HomeViewNotifier extends StateNotifier<HomeViewState> {
...
+        if (session != null && !session.session_active) {
+          state = state.copyWith(isSessionExpired: true);
+        }
+      });
+    } catch (error, stack) {
+      logger.e(
```
Comment (cp-ishita-g) [inline app/lib/ui/flow/home/home_screen_viewmodel.dart:191]: copy error in state

File: app/lib/ui/flow/message/chat/chat_screen.dart [hunk 3]
```diff
@@ -39,38 +39,46 @@ class ChatScreen extends ConsumerStatefulWidget {
 
 class _ChatScreenState extends ConsumerState<ChatScreen> {
   late ChatViewNotifier notifier;
+  late TextEditingController _textController;
```
Comment (cp-ishita-g) [inline app/lib/ui/flow/message/chat/chat_screen.dart:42]: according to out project structure we are creating text controller in view model, remove it form here, space issue in textfield it is not a issue i discussed with senior it is common behaviour, if you can solve it after keeping this var in view model than it is okay, otherwise let it be

File: app/lib/ui/flow/message/chat/chat_view_model.dart [hunk 4]
```diff
@@ -43,12 +41,10 @@ class ChatViewNotifier extends StateNotifier<ChatViewState> {
...
   ChatViewNotifier(this.threadId, this.messageService, this.apiMessageService,
       this.userService, this.currentUser)
-      : super(ChatViewState(
-            message: TextEditingController(),
-            currentUserId: currentUser?.id ?? ''));
+      : super(ChatViewState(currentUserId: currentUser?.id ?? ''));
 
   void onChange(String text) {
-    state = state.copyWith(allowSend: text.trim().isNotEmpty);
+    state = state.copyWith(allowSend: text.isNotEmpty);
```
Comment (cp-ishita-g) [inline app/lib/ui/flow/message/chat/chat_view_model.dart:47]: why did you remove trim

File: app/lib/ui/flow/message/chat/chat_screen.dart (hunk 3 above)
Comment (kaushiksaliya) [reply app/lib/ui/flow/message/chat/chat_screen.dart:42]: this is not common behaviour, we are not allow space before the text ask to @cp-radhika-s or megh.

Comment (cp-ishita-g) [reply app/lib/ui/flow/message/chat/chat_screen.dart:42]: we can't handle it from our side it should be user choice if they want to add space before text

File: app/lib/ui/flow/message/chat/chat_view_model.dart (hunk 4 above)
Comment (kaushiksaliya) [reply app/lib/ui/flow/message/chat/chat_view_model.dart:47]: already pass text with trim so not need here.

File: app/lib/ui/flow/message/chat/chat_screen.dart (hunk 3 above)
Comment (cp-megh-l) [reply app/lib/ui/flow/message/chat/chat_screen.dart:42]: Maybe @kaushiksaliya you can check one extra condition before returning boolean value i.e., 

pass un-trimmed value from onChange to viewmodel and check in viewmodel:

if text.isNotEmpty && text.trim().isEmpty() ---> reset text controller and then return
The above condition will pass only if user enters only space characters

File: app/lib/ui/flow/home/home_screen_viewmodel.dart (hunk 1 above)
Comment (kaushiksaliya) [reply app/lib/ui/flow/home/home_screen_viewmodel.dart:185]: no need.

**Review by cp-ishita-g — APPROVED** [review]
